compiler/semmagic.nim

```diff
@@ -326,13 +326,19 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
       # TODO there is no proper way to find out if a type cannot be queried for the size.
       let size = getSize(c.config, n[1].typ)
       # We just assume here that the type might come from the c backend
-      if size > 0:
+      if size == szImportedType:
+        # Forward to the c code generation to emit a `sizeof` in the C code.
+        result = n
+      elif size >= 0:
```

Comment:
will that work with incomplete types that have at least 1 field? 
```
type
  MyStruct {.importc: "MyStruct".} = object
    field1: int
    # missing field2
```

see also https://github.com/nim-lang/Nim/issues/9250 for a more robust solution (that can be implemented after your PR)

Reply:
originally I was looking at the wrong location for the importc flag. Now it should work properly.

Reply:
@krux02 is my assessment in https://github.com/nim-lang/Nim/issues/9250 correct that currently you must pessimistically reject computing sizeof for any type annotated as `importc` ? feel free to comment in https://github.com/nim-lang/Nim/issues/9250

Reply:
you are correct.